joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/InheritanceFullNamePass.scala

```diff
@@ -1,39 +1,49 @@
 package io.joern.pysrc2cpg
 
+import io.joern.x2cpg.passes.base.TypeDeclStubCreator
 import io.shiftleft.codepropertygraph.Cpg
-import io.shiftleft.codepropertygraph.generated.nodes.{Call, TypeDecl}
+import io.shiftleft.codepropertygraph.generated.nodes.{Call, TypeDecl, TypeDeclBase}
 import io.shiftleft.codepropertygraph.generated.{EdgeTypes, PropertyNames}
 import io.shiftleft.passes.CpgPass
 import io.shiftleft.semanticcpg.language._
-import overflowdb.BatchedUpdate
 
-import java.util.regex.Pattern
+import java.io.File
+import java.util.regex.{Matcher, Pattern}
+import scala.collection.mutable
 
 /** Using some basic heuristics, will try to resolve type full names from types found within the CPG. Requires
   * ImportPass as a pre-requisite.
   */
 class InheritanceFullNamePass(cpg: Cpg) extends CpgPass(cpg) {
-  override def run(builder: BatchedUpdate.DiffGraphBuilder): Unit =
+
+  private val typeDeclMap =
+    mutable.HashMap.from[String, TypeDeclBase](cpg.typeDecl.where(_.nameNot(".*<.*>$")).map(t => t.fullName -> t))
+
+  override def run(builder: DiffGraphBuilder): Unit = {
     cpg.typeDecl
       .filterNot(t =>
         t.inheritsFromTypeFullName == Seq("ANY") || t.inheritsFromTypeFullName == Seq(
           "object"
         ) || t.inheritsFromTypeFullName.isEmpty
       )
       .foreach { t =>
-        val resolvedTypeDecls = resolveInheritedTypeFullName(t)
-        builder.setNodeProperty(t, PropertyNames.INHERITS_FROM_TYPE_FULL_NAME, resolvedTypeDecls.map(_.fullName))
-        cpg.typ
-          .fullNameExact(resolvedTypeDecls.fullName.toSeq: _*)
-          .foreach(tgt => builder.addEdge(t, tgt, EdgeTypes.INHERITS_FROM))
+        val resolvedTypeDecls = resolveInheritedTypeFullName(t, builder)
+        if (resolvedTypeDecls.nonEmpty) {
+          val fullNames = resolvedTypeDecls.map(_.fullName)
+          builder.setNodeProperty(t, PropertyNames.INHERITS_FROM_TYPE_FULL_NAME, fullNames)
+          cpg.typ
```

Comment:
Seems like a potential bottleneck. Maybe initialize a map to increase lookup speed here?